src/vss_tools/vspec/vssexporters/vss2samm/helpers/ttl_helper.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2024 Contributors to COVESA
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License 2.0 which is available at
+# https://www.mozilla.org/en-US/MPL/2.0/
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from pathlib import Path
+
+from rdflib import Graph, URIRef
+from rdflib.namespace import RDF
+from vss_tools import log
+from vss_tools.vspec.model import NodeType
+from vss_tools.vspec.tree import VSSNode
+
+from ..config import config as cfg
+from . import ttl_builder_helper as ttl_builder
+from . import vss_helper as vss_helper
+from .file_helper import write_graph_to_file
+from .namespaces import get_node_name_from_vspec_uri, samm_output_namespace
+from .samm_concepts import SammConcepts, VSSConcepts
+from .string_helper import str_to_uc_first_camel_case
+
+
+# Parse provided VSSNode to an RDF Graph and write it to a TTL file
+def parse_vss_tree(path_to_ttl: Path, vss_node: VSSNode, split_vss: bool):
+    # Process provided vss_node so to get all of its unique VSSNode names, check if there is any duplicates etc.
+    # This will be further used to make sure that we don't get any duplicate nodes in the generated TTL graph(s)
+    vss_helper.count_vss_tree_unique_node_names(vss_node)
+
+    log.debug(
+        "Parse VSS node: '%s' to TTL file\n  -- as aspect%s\n", vss_node.name, "\n  -- split" if split_vss else ""
+    )
+
+    # Skip parsing of VSSNode is branches, which have only deprecated children
+    # NOTE: this is a special case for the OBD branch, which children are marked as deprecated from VSS 5.0,
+    #       but the branch itself is not marked as deprecated.
+    deprecated_children = []
+    if hasattr(vss_node.data, "type") and vss_node.data.type is NodeType.BRANCH:
```

Comment:
every vssnode has `type`

Reply:
I got this error, when I removed the hasattr check:
![2024-09-05_14h38_14](https://github.com/user-attachments/assets/6d5a427c-06ae-4fe6-9c29-77567c9a0211)